### DISCUSSION TITLE
Is AMD still supported under Windows?

### DISCUSSION THREAD

**Discussion by OldPixelReaper** [vladmandic/sdnext#874] [Q&A] [answered]
### Is AMD still supported under Windows?

AMD RX 5700 XT, 32 GB RAM under Windows 11, current drivers. 
Hello, everything worked for me 2 weeks ago, however, for a week it seems to run only on the CPU. I deleted everything today, reinstalled everything again completely. 

But still extremely slow. I think that recently it no longer recognizes my graphics card. Has anything changed, do I need to configure, install something? Out of the box it no longer works for me. I don't know much about Python, but with some help I'll muddle through if anyone has a solution. I have 24 it/s , normally that is 1 it/s.
With a different UI it works.
[startup.txt](https://github.com/vladmandic/automatic/files/11454963/startup.txt)
Thank you and sorry for the bumpy English

**Comment by vladmandic** [comment]
AMD GPUs were never supported on Windows as AMD never released ROCm package for Windows, only Linux.
There are talks that AMD will release it soon. So not sure what were you using before and had GPU utilized under Windows?

**Reply by OldPixelReaper** [reply]
I don't understand?

"Note: AMD and M1 platforms are supported, but without out-of-the-box optimizations.

https://github.com/vladmandic/automatic/tree/directml-stable

That worked 2 weeks ago didn't it?

**Reply by vladmandic** [reply, marked as answer]
Yes, you can still run using DirectML. That's not what I consider "native" GPU, but yes, it will use GPU.
But its not enabled by default (it was just for a day and it shouldn't be) - you need to run with `--use-directml` flag.
I see you already have `--upgrade` and `--medvram`, just add that one as well.

**Reply by OldPixelReaper** [reply]
Ahhhh. Ok, I see, I got exactly the build where the enabled was. Yes, native was expressed wrong, I'm not really good in english. Ok, now I have

20:31:00-730236 INFO Using DirectML Backend

20:31:02-195998 INFO Torch 2.0.0+cpu

But it is still as slow. Do I have to set anything else? thx 4 support
[startup2.txt](https://github.com/vladmandic/automatic/files/11455403/startup2.txt)

**Comment by vladmandic** [comment]
it looks like DirectML is returning some incorrect results in FP16 precision mode, so app now forces it into slower FP32 mode.
that check didn't exist before, but I've added it as safety item as soo many users were reporting "black images" or NaN errors.
need to investigate why is DirectML reporting FP16 precision errors and then I can perhaps re-enable it.

**Reply by OldPixelReaper** [reply]
I understand, thanks
